walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,140 @@
 
 package com.google.sps;
 
-import java.util.Collection;
+import java.util.*;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    if (request == null) {
+      return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+
+    if (events == null || events.size() <= 0) {
+      if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+        return new ArrayList<>();
+      } else {
+        return Arrays.asList(TimeRange.WHOLE_DAY);
+      }
+    }
+
+    Collection<String> attendees = request.getAttendees();
```

Comment:
should you use Set here since you keep iterating on them to compare. You can use `LinkedHashSet` to preserve the order if needed.

Reply:
I think what Yohan means is that instead of what you're doing now: for each event, get list of event attendees, check if they overlap with my list of attendee by looping through two list (note that ArrayList.contains is O(N)), so effectively you're doing something O(N^2).

Yohan suggests a set approach (note that some set, e.g hashSet has contains in O(1)).